tedana/model/combine.py

```diff
@@ -3,11 +3,94 @@
 """
 import logging
 import numpy as np
-from tedana import utils
+from tedana.utils import unmask
+from tedana.due import due, BibTeX, Doi
 
 LGR = logging.getLogger(__name__)
 
 
+@due.dcite(BibTeX('@article{doi:10.1002/(SICI)1522-2594(199907)42:1<87::AID-'
+                  'MRM13>3.0.CO;2-O,'
+                  'author = {Posse Stefan and Wiese Stefan and Gembris Daniel '
+                  'and Mathiak Klaus and Kessler Christoph and Grosse-Ruyken '
+                  'Maria-Liisa and Elghahwagi Barbara and Richards Todd and '
+                  'Dager Stephen R. and Kiselev Valerij G.},'
+                  'title = {Enhancement of BOLD-contrast sensitivity by '
+                  'single-shot multi-echo functional MR imaging},'
+                  'journal = {Magnetic Resonance in Medicine},'
+                  'volume = {42},'
+                  'number = {1},'
+                  'pages = {87-97},'
+                  'doi = {10.1002/(SICI)1522-2594(199907)42:1<87::AID-MRM13>'
+                  '3.0.CO;2-O},'
+                  'url = {https://onlinelibrary.wiley.com/doi/abs/10.1002/'
+                  '%28SICI%291522-2594%28199907%2942%3A1%3C87%3A%3AAID-'
+                  'MRM13%3E3.0.CO%3B2-O}}'),
+           description='T2* method of combining data across echoes using '
+                       'monoexponential equation.')
+def _combine_t2s(mdata, tes, ft2s):
+    """
+    Combine data across echoes using weighted averaging according to voxel-
+    (and sometimes volume-) wise estimates of T2*.
+
+    Parameters
+    ----------
+    mdata : (S x E x T) array_like
+        Masked data.
+    tes : (E,) array_like
+        Echo times in milliseconds.
+    ft2s : (S [x T]) array_like
+        Either voxel-wise or voxel- and volume-wise estimates of T2*.
+
+    Returns
+    -------
+    combined : (S x T) :obj:`numpy.ndarray`
+        Data combined across echoes according to T2* estimates.
+    """
+    n_vols = mdata.shape[-1]
+    alpha = tes * np.exp(-tes / ft2s)
+    if alpha.ndim == 2:
+        # Voxel-wise T2 estimates
+        alpha = np.tile(alpha[:, :, np.newaxis], (1, 1, n_vols))
+    elif alpha.ndim == 3:
+        # Voxel- and volume-wise T2 estimates
+        # alpha is currently (S, T, E) but should be (S, E, T) like mdata
+        alpha = np.swapaxes(alpha, 1, 2)
+
+        # If all values across echos are 0, set to 1 to avoid
+        # divide-by-zero errors
+        ax0_idx, ax2_idx = np.where(np.all(alpha == 0, axis=1))
+        alpha[ax0_idx, :, ax2_idx] = 1.
+    combined = np.average(mdata, axis=1, weights=alpha)
+    return combined
+
+
+@due.dcite(Doi('10.1002/mrm.20900'),
```

Comment:
Should we expand this out with the author list and title ?

Reply:
I can add the first author and year to the description, but as long as duecredit can find the DOI we don't need all of the BibTeX info. My concern with the T2* citation was that the DOI looked weird, but after testing it that one rendered too, so I can switch out the clunky BibTeX citation for just the DOI.

Reply:
Got it, thanks for clarifying ! Yes, being consistent either with only the DOI or with the whole BibTeX citation sounds great to me :smile: